File: python/_1_srp_single_responsibility_principle/apps/GradeCalculater.py
from .Student import Student


class GradeCalculater:
    def __init__(self, fail_grade: float = 59.99):
        self.fail_grade = fail_grade

    def average(self, student: Student) -> float:
        grades = student.get_grades()
        return sum(grades) / len(grades)
    
    def status(self, student: Student) -> str:
        return 'Pass' if self.average(student=student) > self.fail_grade else 'Fail'

File: javascript/_1_srp_single_responsibility_principle/classess/GradeCalculater.js


export class GradeCalculater {
    
    constructor(failGrade = 59.99) {
        this.failGrade = failGrade;
    }

    average(student) {
        const grades = student.getGrades();
        const sum = grades.reduce((acc, grade) => acc + grade, 0);
        return sum / grades.length;
    }

    status(student) {
        const averageGrade = this.average(student);
        return averageGrade >= this.failGrade ? "Passed" : "Failed";
    }

}